sources/TileDB.CSharp/Marshalling/SafeHandles/ArrayHandle.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Runtime.InteropServices;
+using TileDB.Interop;
+
+namespace TileDB.CSharp.Marshalling.SafeHandles
+{
+    internal unsafe sealed class ArrayHandle : SafeHandle
+    {
+        public ArrayHandle() : base(IntPtr.Zero, true) { }
+
+        public ArrayHandle(IntPtr handle, bool ownsHandle) : base(IntPtr.Zero, ownsHandle) { SetHandle(handle); }
+
+        public static ArrayHandle CreateUnowned(tiledb_array_t* array) => new((IntPtr)array, ownsHandle: false);
+
+        public static ArrayHandle Create(Context context, sbyte* uri)
+        {
+            var handle = new ArrayHandle();
+            var successful = false;
+            tiledb_array_t* array = null;
+            try
+            {
+                using (var ctx = context.Handle.Acquire())
+                {
+                    context.handle_error(Methods.tiledb_array_alloc(ctx, uri, &array));
+                }
+                successful = true;
+            }
+            finally
+            {
+                if (successful)
+                {
+                    handle.InitHandle(array);
+                }
+            }
+
+            return handle;
+        }
```

Comment:
We mimic the way the .NET runtime marshals safe handles from native code. Instead of just a `new ArrayHandle((IntPtr)array, ownsHandle: true)`, we have to first create it with an invalid value and then assign it before we return it.

![image](https://user-images.githubusercontent.com/12659251/196117779-ae5d95d0-dc69-4920-8b9a-763fd1027fac.png)